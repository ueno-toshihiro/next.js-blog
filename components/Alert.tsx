import React from "react";
import styled, { StyledComponent } from 'styled-components';

const Success: StyledComponent<"div", any, {}, never> = styled.div`
  color: green;
`
const Error: StyledComponent<"div", any, {}, never> = styled.div`
  color: red;
`

const ResElement = ({children, type}): JSX.Element => {
  if (type === 'success') {
    return <Success>{children}</Success>
  }

  return <Error>{children}</Error>
}

type AlertPropsType = {
  children: React.ReactNode;
  type: 'success' | 'error';
};

export default function Alert({ children, type }: AlertPropsType): JSX.Element {
  return <ResElement type={type}>{children}</ResElement>
}
