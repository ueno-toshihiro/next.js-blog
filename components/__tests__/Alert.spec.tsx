import React from 'react';
import renderer from 'react-test-renderer';
import Alert from '../Alert';

test('AppTitle', () => {
  const component = renderer.create(<Alert type="success" >成功！</Alert>);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
