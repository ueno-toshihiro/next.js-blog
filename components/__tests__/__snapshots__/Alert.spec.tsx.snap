// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTitle 1`] = `
<div
  className="sc-bdfBwQ gkBfKD"
>
  成功！
</div>
`;
